dist/tools/firmware_metadata/Makefile

```diff
@@ -0,0 +1,19 @@
+RIOTBASE := ../../..
+RIOT_INCLUDE = $(RIOTBASE)/sys/include
+SHA256_DIR := $(RIOTBASE)/sys/hashes
+SHA256_INCLUDE := $(RIOT_INCLUDE)/hashes
+METADATA_SRC := generate-metadata.c $(SHA256_DIR)/sha256.c
+METADATA_HDR := $(RIOT_INCLUDE)/hashes/sha256.h
+
+CFLAGS += -g -O3 -Wall -Wextra -pedantic -std=c99
+
+all: bin/generate-metadata
+
+bin/:
+	mkdir -p bin
+
+bin/generate-metadata: bin/ $(METADATA_HDR) $(METADATA_SRC)
```

Comment:
When a file is changed, then the modification timestamp of its parent folder gets updated. A newer timestamp of a requirement $X tells Make to rebuild $X. So updating $X will make $X look outdated to Make. Every time you run Make it will update $X, which wastes a tiny amount of wall clock time.

As a rule of thumb directories should always be prerequirements, not requirements, unless you are doing something really odd. :)